File: rxoauth/src/androidTest/java/cz/ackee/rxoauth/OAuthStoreTest.java
package cz.ackee.rxoauth;


import android.content.Context;
import android.content.SharedPreferences;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static android.support.test.InstrumentationRegistry.getTargetContext;
import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNull;

/**
 * Test for oauth store
 */
@RunWith(JUnit4.class)
public class OAuthStoreTest {

    @Before
    public void clean() {
        OAuthStore store = new OAuthStore(getTargetContext());
        store.onLogout();
        store = new OAuthStore(getTargetContext().getSharedPreferences("sp", Context.MODE_PRIVATE));
        store.onLogout();
    }

    public void testSaveOauthCredentialsContextConstructor() throws Exception {
        OAuthStore store = new OAuthStore(getTargetContext());
        store.saveOauthCredentials("a", "b");
        assertEquals(store.getAccessToken(), "a");
    }

    @Test
    public void testGetAccessToken() throws Exception {
        OAuthStore store = new OAuthStore(getTargetContext());
        store.saveOauthCredentials("abc", "def");
        assertEquals(store.getAccessToken(), "abc");
    }

    @Test
    public void testGetRefreshToken() throws Exception {
        OAuthStore store = new OAuthStore(getTargetContext());
        store.saveOauthCredentials("abc", "def");
        assertEquals(store.getRefreshToken(), "def");
    }

    @Test
    public void testOnLogout() throws Exception {
        OAuthStore store = new OAuthStore(getTargetContext());
        store.saveOauthCredentials("abc", "def");
        assertEquals(store.getAccessToken(), "abc");
        store.onLogout();
        assertEquals(store.getAccessToken(), null);
    }

    @Test
    public void testSaveOauthCredentials1() throws Exception {
        SharedPreferences sp = getTargetContext().getSharedPreferences("sp", Context.MODE_PRIVATE);
        assertNull(sp.getString("oath2_access_token", null));
        OAuthStore store = new OAuthStore(sp);
        store.saveOauthCredentials("a", "b");
        assertEquals(store.getAccessToken(), sp.getString("oath2_access_token", null));
        // clean
    }
}

File: rxoauth/src/androidTest/java/cz/ackee/rxoauth/OAuthStoreTest.kt
package cz.ackee.rxoauth

import android.content.Context
import android.support.test.InstrumentationRegistry.getTargetContext
import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertNull
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

/**
 * Test for oauth store
 */
@RunWith(JUnit4::class)
class OAuthStoreTest {

    @Before
    fun clean() {
        var store = OAuthStore(getTargetContext())
        store.onLogout()
        store = OAuthStore(getTargetContext().getSharedPreferences("sp", Context.MODE_PRIVATE))
        store.onLogout()
    }

    @Throws(Exception::class)
    fun testSaveOauthCredentialsContextConstructor() {
        val store = OAuthStore(getTargetContext())
        store.saveOauthCredentials("a", "b")
        assertEquals(store.accessToken, "a")
    }

    @Test
    @Throws(Exception::class)
    fun testGetAccessToken() {
        val store = OAuthStore(getTargetContext())
        store.saveOauthCredentials("abc", "def")
        assertEquals(store.accessToken, "abc")
    }

    @Test
    @Throws(Exception::class)
    fun testGetRefreshToken() {
        val store = OAuthStore(getTargetContext())
        store.saveOauthCredentials("abc", "def")
        assertEquals(store.refreshToken, "def")
    }

    @Test
    @Throws(Exception::class)
    fun testOnLogout() {
        val store = OAuthStore(getTargetContext())
        store.saveOauthCredentials("abc", "def")
        assertEquals(store.accessToken, "abc")
        store.onLogout()
        assertEquals(store.accessToken, null)
    }

    @Test
    @Throws(Exception::class)
    fun testSaveOauthCredentials1() {
        val sp = getTargetContext().getSharedPreferences("sp", Context.MODE_PRIVATE)
        assertNull(sp.getString("oath2_access_token", null))
        val store = OAuthStore(sp)
        store.saveOauthCredentials("a", "b")
        assertEquals(store.accessToken, sp.getString("oath2_access_token", null))
        // clean
    }
}